### ISSUE TITLE
Coresets: Do forward pass once instead of twice

### ISSUE THREAD

**Issue by francescodeaglio** [eth-easl/modyn#219]
In some strategies, points are sampled based on the loss. Typically the workflow is

```
forward_pass(BigBatch)
compute the loss
Small batch = sample from loss distribution
forward_pass(SmallBatch)
backward_pass(SmallBatch)
```

In the original paper, they compute 2 times the forward pass. We need to investigate if something like 
```
big_output = model(BigBatch)
... sample and get indexes ...
loss = loss_fct(big_output[indexes], target[indexes])
loss.backward()
```
Works and is fast. Probably it works but requires the same runtime of doing the backward on all BigBatch samples, so it would make this solution useless
